### DISCUSSION TITLE
gpg: no valid OpenPGP data found. (when adding the ppa)

### DISCUSSION THREAD

**Discussion by ghost** [lah7/gtk3-classic#76] [Help & Support] [answered]
gpg: keybox '/tmp/tmpev0jtpyr/pubring.gpg' created
gpg: /tmp/tmpev0jtpyr/trustdb.gpg: trustdb created
gpg: key 6CFDA3EBE08FDDE9: public key "Launchpad PPA for Luke Horwell" imported
gpg: Total number processed: 1
gpg:               imported: 1
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
gpg: no valid OpenPGP data found.

**Comment by lah7** [comment, marked as answer]
Which distro & version are you using?

**Reply by ghost** [reply]
Debian 11 and XFCE 4.16

**Reply by lah7** [reply]
The PPA is designed for Ubuntu (so it'll have their patches), but it _should_ be somewhat compatible. Not officially supported.

I can't tell what commands you've tried so far, but adding the repository should be something like this on Debian:

```bash
echo "deb http://ppa.launchpad.net/lah7/gtk3-classic/ubuntu jammy main" | sudo tee /etc/apt/sources.list.d/gtk3-classic.list
sudo apt-key adv --recv-key --keyserver keyserver.ubuntu.com 6CFDA3EBE08FDDE9
sudo apt-get update
```

**Reply by lah7** [reply]
The last message looks like Debian 11 might have removed the `apt-key` method, but I don't actually know the successor command yet. If those commands above fail too, please take a look online to see what the alternate solution is:

> Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).

I'm not on Debian to check the manual, but this might note how to do it:

    man apt-key(8)

**Reply by ghost** [reply]
Is it supposed to be the packages with ~classic~?
It does the same thing (as below) with all the ~classic~ packages (the formatting crossed it out because of the ~ ~)
                                   
After this operation, 449 kB of additional disk space will be used.
dpkg-deb: error: archive '/var/cache/apt/archives/libgtk-3-common_3.24.33-2~classic~22.04_all.deb' uses unknown compression for member 'control.tar.zst', giving up
dpkg: error processing archive /var/cache/apt/archives/libgtk-3-common_3.24.33-2~classic~22.04_all.deb (--unpack):
 dpkg-deb --control subprocess returned error exit status 2
Errors were encountered while processing:
 /var/cache/apt/archives/libgtk-3-common_3.24.33-2~classic~22.04_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

**Reply by lah7** [reply]
> dpkg-deb: error: archive '/var/cache/apt/archives/libgtk-3-common_3.24.33-2classic22.04_all.deb' uses unknown compression for member 'control.tar.zst', giving up

Haven't seen that before, but it looks like Launchpad is providing `control.tar.zst` instead of the traditional `control.tar.gz`. At least that's the case for `jammy` (try switching the codename to `impish` or `focal`, though I don't know which GTK version Debian 11 uses). 

[According to this answer](https://unix.stackexchange.com/questions/669004/zst-compression-not-supported-by-apt-dpkg), Apt doesn't support zst compression, but Ubuntu's version does. Instead, what you could do is delete the repository and [download each `*_amd64.deb`](https://launchpad.net/~lah7/+archive/ubuntu/gtk3-classic/+packages) files into a folder and install manually with:

    sudo apt install ./*.deb

**Reply by ghost** [reply]
The focal packages worked. But is there a way to make it so there isn't two title bars? Or is it just my theme that makes it looks like there's two?
![image](https://user-images.githubusercontent.com/89735533/175778207-c48bbd57-849f-4aec-9051-cda86781a028.png)

**Reply by lah7** [reply]
That's a known bug: https://github.com/lah7/gtk3-classic/issues/32

Although you might be able to edit the theme so CSDs look like a toolbar.
